src/processor/reconnection-graph.service.ts

```diff
@@ -266,15 +256,40 @@ export class ReconnectionGraphService {
     const publicFriendships: PaginatedStorageResponse[] = await this.blockchainService.rpc('statefulStorage', 'getPaginatedStorage', dsnpUserId, publicFriendshipSchemaId);
     const privateFollows: PaginatedStorageResponse[] = await this.blockchainService.rpc('statefulStorage', 'getPaginatedStorage', dsnpUserId, privateFollowSchemaId);
     const privateFriendships: PaginatedStorageResponse[] = await this.blockchainService.rpc('statefulStorage', 'getPaginatedStorage', dsnpUserId, privateFriendshipSchemaId);
-
+    const dsnpKeys = await this.formDsnpKeys(dsnpUserId);
+    const graphKeyPairsSdk = graphKeyPairs.map(
+      (keyPair: ProviderKeyPair): GraphKeyPair => ({
+        keyType: GraphKeyType.X25519,
+        publicKey: hexToU8a(keyPair.publicKey),
+        secretKey: hexToU8a(keyPair.privateKey),
+      }),
+    );
     const importBundleBuilder = new ImportBundleBuilder();
     // Only X25519 is supported for now
     // check if all keys are of type X25519
     const areKeysCorrectType = graphKeyPairs.every((keyPair) => keyPair.keyType === KeyType.X25519);
     if (!areKeysCorrectType) {
       throw new Error('Only X25519 keys are supported for now');
     }
-
+    if(dsnpKeys && dsnpKeys.keys.length > 0 && graphKeyPairs && graphKeyPairs.length > 0) {
+      // an empty page is created to store the public keys
+      importBundles.push(
+        importBundleBuilder
+          .withDsnpUserId(dsnpUserId.toString())
+          .withSchemaId(privateFollowSchemaId )
+          .withDsnpKeys(dsnpKeys)
+          .withGraphKeyPairs(graphKeyPairsSdk)
+          .build(),
+      );
+      importBundles.push(
+        importBundleBuilder
+          .withDsnpUserId(dsnpUserId.toString())
+          .withSchemaId(privateFriendshipSchemaId)
+          .withDsnpKeys(dsnpKeys)
+          .withGraphKeyPairs(graphKeyPairsSdk)
+          .build(),
+      );
+    }
```

Comment:
building sort of empty bundle for both private schema_id's so that we can set `graphKeyPairs` sent by provider